### ISSUE TITLE
Use json instead of pickle to save references

### ISSUE THREAD

**Issue by mvdoc** [duecredit/duecredit#112]
Right @yarikoptic ? This could be a nice small item for the hackathon

**Comment by yarikoptic** [comment]
yes!  while at it, probably initiate storing not just into a single file but may be under .duecredit/ ... we should decide though on how to segment/aggregate them later.  E.g. we could trace per each execution and store into a unique dedicated file and then merge whenever summary is requested.  This way we could aggregate later from different runs/environments so they do not conflict over the "ultimate" one

**Comment by shyamd** [comment]
I would like to suggest using the MSONable pattern from [monty](https://github.com/materialsvirtuallab/monty/blob/adbbef8bace52d05665799851c33fc368e3c0f5e/monty/json.py#L35) which gives you an automatic from_dict() and as_dict() for serialization to JSON if your init args are stored unde the same name in your object.

**Comment by clbarnes** [comment]
In particular, CSL-data JSON may be a good fit as it gives you integration with (at least some of) the citeproc ecosystem for free.

**Comment by yarikoptic** [comment]
Some references for CSL and CSL-Data: 
- https://citationstyles.org/
   - https://github.com/citation-style-language/schema 
- https://docs.cloudcite.net/csl

**Comment by yarikoptic** [comment]
Most likely we should just define our relevant classes using https://docs.pydantic.dev/ and provide json or yaml export/import for them.

**Comment by clbarnes** [comment]
I went down the rabbit hole of strong python representations of CSL types some time ago https://github.com/clbarnes/citepy/blob/master/citepy/classes.py but yeah, pydantic's probably a much better choice :sweat_smile: 

In particular, pydantic can generate code from JSON schemas https://docs.pydantic.dev/datamodel_code_generator/ , like those maintained here https://github.com/citation-style-language/schema/tree/master/schemas/input
